onnxruntime/core/graph/featurizers_ops/featurizers_defs.cc

```diff
@@ -215,112 +235,200 @@ void RegisterDateTimeFeaturizerVer1() {
             }
 
             propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_UINT8, 9);
-            if(has_shape) {
+            if (has_shape) {
               propagateShapeFromInputToOutput(ctx, 1, 9);
             }
 
             propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_UINT16, 10);
-            if(has_shape) {
+            if (has_shape) {
               propagateShapeFromInputToOutput(ctx, 1, 10);
             }
 
             propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_UINT16, 11);
-            if(has_shape) {
+            if (has_shape) {
               propagateShapeFromInputToOutput(ctx, 1, 11);
             }
 
             propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_UINT8, 12);
-            if(has_shape) {
+            if (has_shape) {
               propagateShapeFromInputToOutput(ctx, 1, 12);
             }
 
             propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_UINT8, 13);
-            if(has_shape) {
+            if (has_shape) {
               propagateShapeFromInputToOutput(ctx, 1, 13);
             }
 
             propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_UINT8, 14);
-            if(has_shape) {
+            if (has_shape) {
               propagateShapeFromInputToOutput(ctx, 1, 14);
             }
 
             propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_INT32, 15);
-            if(has_shape) {
+            if (has_shape) {
               propagateShapeFromInputToOutput(ctx, 1, 15);
             }
 
             propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_STRING, 16);
-            if(has_shape) {
+            if (has_shape) {
               propagateShapeFromInputToOutput(ctx, 1, 16);
             }
 
             propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_STRING, 17);
-            if(has_shape) {
+            if (has_shape) {
               propagateShapeFromInputToOutput(ctx, 1, 17);
             }
 
             propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_STRING, 18);
-            if(has_shape) {
+            if (has_shape) {
               propagateShapeFromInputToOutput(ctx, 1, 18);
             }
 
             propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_STRING, 19);
-            if(has_shape) {
+            if (has_shape) {
               propagateShapeFromInputToOutput(ctx, 1, 19);
             }
 
             propagateElemTypeFromDtypeToOutput(ctx, ONNX_NAMESPACE::TensorProto_DataType_UINT8, 20);
-            if(has_shape) {
+            if (has_shape) {
               propagateShapeFromInputToOutput(ctx, 1, 20);
             }
+          });
+}
 
-          }
-      );
+void RegisterFromStringFeaturizerVer1() {
+  static const char* doc = R"DOC(
```

Comment:
I would try to avoid such static strings for op documentation as they occupy space in the binary. They're not required when running the model. May be we can disable them behind a macro like ONNX (__ONNX_NO_DOC_STRINGS).

Reply:
Shall we omit SetDoc call?

---
In reply to: [373742554](https://github.com/microsoft/onnxruntime/pull/2958#discussion_r373742554) [](ancestors = 373742554)

Reply:
Unfortunately SetDoc only reduces the memory footprint of the opschema object. The static string is still stuck in the binary occupying space. The ONNX schema is filled with many such examples and so is our contrib ops.

Reply:
I will both Omit SetDoc and make strings comments

---
In reply to: [373746248](https://github.com/microsoft/onnxruntime/pull/2958#discussion_r373746248) [](ancestors = 373746248)